### DISCUSSION TITLE
Bridge support for battery based devices

### DISCUSSION THREAD

**Discussion by timoschilling** [mongoose-os-apps/shelly-homekit#1488] [Ideas]
HomeKit over IP requires a consistent IP connection (WiFi / Ethernet) to work properly, only HomeKit over Bluetooth is made for battery bases devices. To support battery based Shelly's we plan to add a bridge mode to support them. We could also add bridge support for Shelly BLU devices.

This issue replaces #104, #124, #295, #1359.

**Comment by vrpt2022** [comment]
@timoschilling Thanks for this!!!

Do you think #1168 could be included as well here?

**Comment by timoschilling** [comment]
@vrpt2022 yes could be

**Comment by vrpt2022** [comment]
Good morning!

Any update regarding this?

thanks

**Comment by timoschilling** [comment]
no, when you will see it here, no need to ask

**Comment by inaftanaila** [comment]
Hey @timoschilling - thanks for the reply. I bet there's a lot of people interested in this (me included). Is there, currently, any indication of an ETA?

**Comment by timoschilling** [comment]
We don’t have any route map and no ETAs. The enhancement label only means that we think it’s a good idea and that it maybe some day will be implemented. The only thing that I can say that I’m also interested in this feature.

**Comment by caarlos0** [comment]
FWIW I have something working for shelly flood here: https://github.com/caarlos0/homekit-shelly
